src/Avalonia.Controls/Border.cs

```diff
@@ -92,12 +95,63 @@ public override void Render(DrawingContext context)
 
             if (background != null)
             {
-                context.FillRectangle(background, rect, cornerRadius);
+                context.FillRectangle(Brushes.Transparent, rect);
             }
 
-            if (borderBrush != null && borderThickness > 0)
+            if (borderBrush != null)
             {
-                context.DrawRectangle(new Pen(borderBrush, borderThickness), rect, cornerRadius);
+                var pen = new Pen(borderBrush, 4);
+                var resolution = 5.0;
+                
+
+                Point pt(double x, double y)
+                {
+                    return new Point(x, y);
+                }
+
+                void DrawSector(Point pos, double radius, double degStart, double degEnd, double fromSize, double toSize)
+                {
+                    var points = new List<Point>();
+                    var degSteps = ((degEnd - degStart) / resolution);
+                    var sizeStep = (toSize - fromSize) / degSteps/2.0;
+                    var curStep = fromSize / 2.0;
+
+                    for (double i = degStart; i <= degEnd; i += resolution)//outer radius
+                    {
+                        points.Add(new Point(pos.X + Math.Cos(i / 180.0 * Math.PI) * (radius+curStep), pos.Y + (-Math.Sin(i / 180.0 * Math.PI)) * (radius+curStep)));
+                        curStep += sizeStep;
+                    }
+
+                    var geom = new PolylineGeometry(points, true);//fill
```

Comment:
Reuse geometry between drawing calls, tessellation is expensive

Reply:
If I change IsFilled, The Geometry don't require tessellation twice?

Reply:
You are recreating geometry on each `Render` call.

Reply:
I know it. But the easy solution of it is change IsFilled on every draw. Can it invoke recalculations?

Reply:
But now it's not valuable because i am going to rewrite all code